### PR TITLE
Document temporal tables

### LINKED ISSUE #3355: Document temporal tables
See https://github.com/dotnet/efcore/issues/4693

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #3355

### CODE REVIEW

File: entity-framework/core/providers/sql-server/temporal-tables.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+# SQL Server/Azure SQL temporal tables
+
+[SQL Server temporal tables](/sql/relational-databases/tables/temporal-tables) automatically keep track of all data ever stored in a table, even after that data has been updated or deleted. This is achieved by creating a parallel "history table" into which timestamped historical data is stored whenever a change is made to the main table. This allows historical data to be queried, such as for auditing, or restored, such as for recovery after accidental mutation or deletion.
+
+EF Core supports:
+
+* The creation of temporal tables using [EF Core Migrations](xref:core/managing-schemas/migrations/index)
+* Transformation of existing tables into temporal tables, again using Migrations
+* Querying historical data
+* Restoring data from some point in the past
```
Comment (roji) [inline entity-framework/core/providers/sql-server/temporal-tables.md:18]: Is "reading" more accurate than "restoring"? Restoring sounds like the ability to revert the entire table to some past state.

File: entity-framework/core/providers/sql-server/temporal-tables.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        public string Name { get; set; }
+        public string Position { get; set; }
+        public string Department { get; set; }
+        public string Address { get; set; }
+        public decimal AnnualSalary { get; set; }
+    }
+-->
+[!code-csharp[Employee](../../../../samples/core/Miscellaneous/NewInEFCore6/TemporalTablesSample.cs?name=Employee)]
+
+The temporal table created will look like this:
```
Comment (roji) [inline entity-framework/core/providers/sql-server/temporal-tables.md:49]:
```suggestion
The temporal table created looks like this:
```

File: entity-framework/core/providers/sql-server/temporal-tables.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+modelBuilder
+    .Entity<Employee>()
+    .ToTable("Employees", b => b.IsTemporal());
+-->
+[!code-csharp[SimpleConfig](../../../../samples/core/Miscellaneous/NewInEFCore6/TemporalTablesSample.cs?name=SimpleConfig)]
+
+> [!TIP]
+> The code shown here comes from [TemporalTablesSample.cs](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Miscellaneous/NewInEFCore6/TemporalTablesSample.cs).
+
+When using EF Core to create the database, the new table will be configured as a temporal table with the SQL Server defaults for the timestamps and history table. For example, consider an `Employee` entity type:
```
Comment (roji) [inline entity-framework/core/providers/sql-server/temporal-tables.md:34]:
```suggestion
When using EF Core to create the database, the new table is configured as a temporal table with the SQL Server defaults for the timestamps and history table. For example, consider an `Employee` entity type:
```

File: entity-framework/core/providers/sql-server/temporal-tables.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+  Employee Pinky Pie valid from 8/26/2021 4:38:58 PM to 12/31/9999 11:59:59 PM
+  Employee Rainbow Dash valid from 8/26/2021 4:38:58 PM to 12/31/9999 11:59:59 PM
+  Employee Fluttershy valid from 8/26/2021 4:38:58 PM to 12/31/9999 11:59:59 PM
+```
+
+Notice that the `ValidTo` column (by default called `PeriodEnd`) contains the `datetime2` max value. This is always the case for the current rows in the table. The `ValidFrom` columns (by default called `PeriodStart`) contains the UTC time that the row was inserted.
+
+## Querying historical data
+
+EF Core supports queries that include historical data through several new query operators:
```
Comment (roji) [inline entity-framework/core/providers/sql-server/temporal-tables.md:179]:
```suggestion
EF Core supports queries that include historical data through several specialized query operators:
```

File: entity-framework/core/providers/provider-log.md [hunk 5]
```diff
@@ -8,6 +8,9 @@ uid: core/providers/provider-log
 
 # Provider-impacting changes
 
+> [!IMPORTANT]
```
Comment (roji) [inline entity-framework/core/providers/provider-log.md:11]: Should we just remove this page? It seems to have very very little value at this point...

**Review by roji — APPROVED** [review]

File: entity-framework/core/providers/sql-server/temporal-tables.md (hunk 1 above)
Comment (ajcvickers) [reply entity-framework/core/providers/sql-server/temporal-tables.md:18]: Restoring is what I show in the example. That is, reading data from some previous point in time and then updating the current state with that data.

File: entity-framework/core/providers/provider-log.md (hunk 5 above)
Comment (ajcvickers) [reply entity-framework/core/providers/provider-log.md:11]: We should not make it 404. But deleting the content might make sense.

**Pull request merged**